panel/chat/step.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import annotations
+
+from typing import ClassVar, Mapping
+
+import param
+
+from ..io.resources import CDN_DIST
+from ..layout import Card, Row
+from ..pane.image import ImageBase
+from ..pane.markup import HTML, HTMLBasePane, Markdown
+from ..pane.placeholder import Placeholder
+from ..widgets.indicators import BooleanStatus
+from .utils import (
+    avatar_lookup, build_avatar_pane, serialize_recursively, stream_to,
+)
+
+DEFAULT_STATUS_AVATARS = {
+    "pending": BooleanStatus(value=False, margin=0, color="primary"),
+    "running": BooleanStatus(value=True, margin=0, color="warning"),
+    "success": BooleanStatus(value=True, margin=0, color="success"),
+    "failed": BooleanStatus(value=True, margin=0, color="danger"),
+}
+
+
+class ChatStep(Card):
+    collapsed = param.Boolean(
+        default=False,
+        doc="Whether the contents of the Card are collapsed.")
+
+    collapsed_on_success = param.Boolean(
+        default=True,
+        doc="Whether to collapse the card on completion.")
+
+    success_title = param.String(default=None, doc="""
+        Title to display when status is success.""")
+
+    default_avatars = param.Dict(
```

Comment:
I associate the word avatar with a user, here it's used for the status. Not sure what I would suggest yet. Also docstring probably has to list the valid status values.